tensorflow_datasets/core/example_serializer.py

```diff
@@ -69,9 +69,12 @@ def _dict_to_tf_example(example_dict, tensor_info_dict):
     example_proto: `tf.train.Example`, the encoded example proto.
   """
   def run_with_reraise(fn, k, example_data, tensor_info):
-    with utils.try_reraise(
```

Comment:
Thank you for fixing this. 
Is the contextmanager really the bottleneck ? I would assume it to add a negligeable overhead (~5 s / millions examples)

The f-string however might slow down things. How about updating `try_reraise` instead ?

```
utils.try_reraise('{} {}', k, tensor_info)
```

And do the `.format` inside the `except` block ?

What do you think ?

Reply:
I agree that moving `.format` inside the except block would improve performance, but I think the context manager is also slowing down the performance significantly. I did a timed run after moving `.format` into the except block and kept the context manager. The resulting runtime is 7m25.011s, which is slower than a timed run with the version without the context manager, which took 6m30.327s.